.circleci/config.yml

```
@@ -296,19 +296,12 @@ defaults:
 
   - base_osx: &base_osx
       macos:
-        xcode: "13.2.0"
+        xcode: "14.2.0"
```

I see that CircleCI offers 14.3.0, which runs on macOS 13 (14.2.0 is still on macOS 12). We should use the latest version, unless there's a good reason not to. I'm not aware if there are any problems with that, but I'd at least try to use it and see.

```suggestion
        xcode: "14.3.0"
```

---

So in other words,  base_osx should have the macos.x86.medium.gen2, and the xcode version should be 14.3.0. Also the base_osx_large should go back to using the large resource tag?

---

> So in other words, base_osx should have the macos.x86.medium.gen2, and the xcode version should be 14.3.0. Also the base_osx_large should go back to using the large resource tag?

Yes :)

---

It failed at b_osx again. It says job was rejected because resource class large, image xcode:14.3.0 is not a valid resource class. What version of xcode should I use @r0qs ?

---

Hi @BraxWong sorry for the late reply. I just looked again to the [CircleCI's macOS Resource Deprecation Announcement](https://discuss.circleci.com/t/macos-resource-deprecation-update/46891) and noticed this:
```
Phase 1 - mid-March 2023
a. Xcode 14.2.0 is the last Xcode version that will be supported by medium and large.
b. We will automatically upgrade jobs for customers on our Free plan to our Gen2 resources (macos.x86.medium.gen2).
```
So it will not be possible to use `14.3.0` as @cameel suggested for the image using `large` resource class. 
Please, change both to `14.2.0` and we should be good to go. And remember to squash the commits ;)

---

I don't really understand how to use ```git rebase -i HEAD~3``` either. I know I have to the recent commits to the old ones but I kept getting ```error: cannot 'squash' without a previous commit You can fix this with 'git rebase --edit-todo' and then run 'git rebase --continue'. Or you can abort the rebase with 'git rebase --abort'``` error. Also a lot of tests have failed after I switched back to 14.2.0 @r0qs

---

No problem. I fixed for you. Thanks. You can read more about the git rebase [here](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) and you can also read about [squashing commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History), those will be very useful for future contributions ;)